### PR TITLE
Turn current page into the default Cart/Checkout page

### LINKED ISSUE #6864: Update default Cart/Checkout notice to turn the current page into the default cart/checkout one
Currently, when you open a page that has Cart or Checkout block, and that page isn't the default Cart/Checkout page, you will get a notice telling you to update by taking you to the settings page:

> If you would like to use this block as your default checkout you must update your <a>page settings in WooCommerce</a>

We want to change that so you can make it the default without needing to leave the editor.

To reduce confusing, we will also switch the current page URL to `/cart` or `/checkout` respectively, or reuse the previous page URL.
image
image

### LINKED ISSUE #6865: Show default Cart/Checkout notice on all inner blocks
This is a follow up to this issue https://github.com/woocommerce/woocommerce-blocks/issues/6864

Currently, the notice is only visible on the top level Cart/Checkout, which might make discoverability a bit hard, we want to show it on all blocks.

### PR DESCRIPTION
**Pull request by senadir**
This PR adds a "make current page default cart/checkout", it has a long flow that goes via several conditions:

To test this PR, you need this PR from to have the latest trunk of WooCommerce, or WooCommerce 6.9.0
1. The notice is visible on Cart/Checkout and all of their inner blocks.
2. If the page isn't saved and published, it will get published, this is a condition in order to make it the default cart/checkout page.
3. Once clicked, the current page is made to be the default cart/checkout page.
4. The current page permalink will change to whatever the previous Cart/Checkout link was, so if that link was /cart the current page is /cart, if it was /carro, so will the current page.
5. If we didn't have a previous page, then it will just use /cart or /checkout like WooCommerce core.
6. The previous page link will get -2 appended to it to avoid conflict.
7. The page is saved.
8. The notice is now green, you can dismiss it.

For older versions of WC (pre 6.9.0), the notice would show the old text
image

Fixes #6864 
Fixes #6865 

### Screenshots

https://user-images.githubusercontent.com/6165348/183219991-4b904273-67ef-43e1-94fb-7498531f104f.mov

#### User Facing Testing

1. Make sure you have a Cart and Checkout pages in WooCommerce -> Settings -> Advanced.
2. Those pages urls can be `/cart` and `/checkout`.
3. Open a new page, give it a title, insert Cart block.
4. Select Cart block, you should see the notice, select an inner block, you should see the notice as well.
5. Click on the notice, it will load for a couple of seconds, once it's green, visit your new page.
6. Make sure it's on the previous url, so `/cart`.
7. Go back to settings, make sure your new page is that value there.
8. Do the same thing for Checkout block.
9. Open a new page, add Cart block, save the page and publish it.
10. Select the block or one of its inner blocks, start the flow again.
11. It should work, confirm that by visiting the frontend.

### WooCommerce Visibility

* [x] WooCommerce Core
* [x] Feature plugin
* [ ] Experimental

### Changelog

> Allow making the Cart/Checkout block page the default one from within the editor.

### CODE REVIEW

File: assets/js/blocks/cart-checkout-shared/sidebar-notices/index.tsx [hunk 1]
```diff
@@ -66,10 +61,10 @@ const withSidebarNotices = createHigherOrderComponent(
 			<>
 				{ ( isCart || isCheckout ) && (
 					<InspectorControls>
-						<CartCheckoutSidebarCompatibilityNotice
+						<DefaultNotice
 							block={ isCheckout ? 'checkout' : 'cart' }
 						/>
-						<DefaultNotice
+						<CartCheckoutSidebarCompatibilityNotice
```
Comment (senadir) [inline assets/js/blocks/cart-checkout-shared/sidebar-notices/index.tsx]: Switched to follow design.

File: assets/js/editor-components/default-notice/index.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	const noticeContent =
+		block === 'checkout'
+			? __(
+					'If you would like to use this block as your default checkout, update your page settings',
+					'woo-gutenberg-products-block'
+			  )
+			: __(
+					'If you would like to use this block as your default cart, update your page settings',
+					'woo-gutenberg-products-block'
+			  );
```
Comment (senadir) [inline assets/js/editor-components/default-notice/index.tsx:40]: We handle Cart and Checkout from this component, everything is the same we just need to have 2 different values.

File: assets/js/editor-components/sidebar-compatibility-notice/index.tsx [hunk 3]
```diff
@@ -33,16 +34,14 @@ export const CartCheckoutSidebarCompatibilityNotice = ( {
...
+				{ 'is-hidden': ! isVisible },
+			] ) }
 		>
-			<Notice
-				onRemove={ dismissNotice }
-				className={ 'wc-blocks-sidebar-compatibility-notice__notice' }
-			>
-				{ noticeText }
-			</Notice>
-		</div>
```
Comment (senadir) [inline assets/js/editor-components/sidebar-compatibility-notice/index.tsx:46]: Removed the extra div.

**Comment by ralucaStan** [comment]
Based on the internal conversation p1660321370894249/1660311130.074619-slack-C8X6Q7XQU we will need to change the implementation: 
- for perior to WC 6.8.0 we just show the current (old) notice that redirects you to the settings page. 
- for WC >= 6.8.0 we show the new notification with the button, that the PR implements

File: assets/js/blocks/cart-checkout-shared/sidebar-notices/index.tsx [hunk 4]
```diff
@@ -66,12 +64,19 @@ const withSidebarNotices = createHigherOrderComponent(
...
+						{ isWcVersion( '6.9.0', '>=' ) ? (
+							<DefaultNotice
+								block={ isCheckout ? 'checkout' : 'cart' }
+							/>
+						) : (
+							<LegacyNotice
+								block={ isCheckout ? 'checkout' : 'cart' }
+							/>
+						) }
+
```
Comment (senadir) [inline assets/js/blocks/cart-checkout-shared/sidebar-notices/index.tsx:76]: This logic should handle older versions which doesn't support the setting needed.

**Comment by tarunvijwani** [comment]
@senadir Thank you for working on it. If I add a Checkout block and use a button to update the settings, it changes the slug to `Cart`. 


https://user-images.githubusercontent.com/11503784/185606335-c97d55bb-1727-45fe-8325-97a35d281511.mov


For the Cart page, it is not updating the slug for me:

https://user-images.githubusercontent.com/11503784/185606425-d6069151-e166-4e75-8d3f-c6514dcdae61.mov

**Comment by senadir** [comment]
Can you please remove any existing saved Cart/Checkout page settings you have, it seems your previously default Checkout page has a `cart` path, and this is what's causing the issue.

**Review by tarunvijwani — APPROVED** [review]
Works great! Let's 🚢  it. 👍

**Comment by tarhi-saad** [comment]
Thank you for your review, @tarunvijwani! I'm merging this PR to include it in the 8.4.0 release!

**Pull request merged**